tests/src/performance/Scenario/JitBench/Benchmarks/WebAppBenchmarks.cs

```diff
@@ -6,52 +6,66 @@
+            ExePath = ExecutableName;
+        }
+
+        protected abstract string ExecutableName { get; }
 
         public override async Task Setup(DotNetInstallation dotNetInstall, string outputDir, bool useExistingSetup, ITestOutputHelper output)
         {
             if(!useExistingSetup)
             {
                 using (var setupSection = new IndentedTestOutputHelper("Setup " + Name, output))
                 {
-                    await DownloadAndExtractJitBenchRepo(outputDir, setupSection);
+                    await CloneAspNetJitBenchRepo(outputDir, setupSection);
                     await CreateStore(dotNetInstall, outputDir, setupSection);
                     await Publish(dotNetInstall, outputDir, setupSection);
                 }
             }
-            string musicStoreSrcDirectory = GetMusicStoreSrcDirectory(outputDir);
+            string webAppSrcDirectory = GetWebAppSrcDirectory(outputDir);
             string tfm = DotNetSetup.GetTargetFrameworkMonikerForFrameworkVersion(dotNetInstall.FrameworkVersion);
-            ExePath = "MusicStore.dll";
-            WorkingDirPath = GetMusicStorePublishDirectory(dotNetInstall, outputDir, tfm);
-            EnvironmentVariables.Add("DOTNET_SHARED_STORE", GetMusicStoreStoreDir(outputDir));
+            WorkingDirPath = GetWebAppPublishDirectory(dotNetInstall, outputDir, tfm);
+            EnvironmentVariables.Add("DOTNET_SHARED_STORE", GetWebAppStoreDir(outputDir));
         }
 
-        async Task DownloadAndExtractJitBenchRepo(string outputDir, ITestOutputHelper output)
+        async Task CloneAspNetJitBenchRepo(string outputDir, ITestOutputHelper output)
         {
             // If the repo already exists, we delete it and extract it again.
             string jitBenchRepoRootDir = GetJitBenchRepoRootDir(outputDir);
             FileTasks.DeleteDirectory(jitBenchRepoRootDir, output);
 
-            string localJitBenchRepo = GetLocalJitBenchRepoDirectory();
-            if (localJitBenchRepo == null)
-            {
-                var url = $"{JitBenchRepoUrl}/archive/{JitBenchCommitSha1Id}.zip";
-                FileTasks.DeleteDirectory(jitBenchRepoRootDir + "_temp", output);
-                await FileTasks.DownloadAndUnzip(url, jitBenchRepoRootDir+"_temp", output);
-                FileTasks.MoveDirectory(Path.Combine(jitBenchRepoRootDir + "_temp", $"JitBench-{JitBenchCommitSha1Id}"), jitBenchRepoRootDir, output);
-            }
-            else
-            {
-                if (!Directory.Exists(localJitBenchRepo))
-                {
-                    throw new Exception("Local JitBench repo " + localJitBenchRepo + " does not exist");
-                }
-                FileTasks.DirectoryCopy(localJitBenchRepo, jitBenchRepoRootDir, output);
-            }
+            // clone the repo, checkout the commit and update submodule
+            await new ProcessRunner("git", $"clone {JitBenchRepoUrl} {jitBenchRepoRootDir}").WithLog(output).Run();
```

Comment:
Does `ProcessRunner.Run` throw exception if failure? We should not continue running when there is a failure.

Reply:
@jorive very good point! I just checked the code and the method returns `Task<int>` where the `int` is processes exit code.  I need to think about the error handling because using the process exit code might not be enough. Some tools when working OK but having some warning returns exit code != 0.